docs/concepts/functional-components.md

```diff
@@ -0,0 +1,21 @@
+---
+title: Functional Components
+sidebar_label: Introduction
+description: Introduction to functional components 
+---
+
+:::note
+Functional components are currently unreleased.
+:::
+
+Functional components are simply Rust functions which act like Yew components. 
+
+## Creating functional components
+
+Functional components are created using the [`#[functional_component]`](functional-components/macro.md) attribute macro.
```

Comment:
All the documentation you've written so far (including the docstring) doesn't really tell the user what's happening under the hood. Could you please add a few sentences explaining the `FunctionProvider` trait and the `FunctionComponent` struct just to give some context. There are cases where you'd have to implement it manually and giving the user a general idea of what an attribute actually does never hurts.

Reply:
What are those cases?

I think doing this is similar to documenting how `html!` uses `VNode`. If this is something important, it can be added to the docstring. I'm not a fan of adding implemention details to website though.

Reply:
`FunctionProvider` isn't not just an implementation detail. It's part of the public API surface (for better or worse).
By giving an overview of how things work it's easier to explain why certain restrictions exist and what to watch out for. It's also valuable for potential contributors because they already have an idea of how the magic works.


> I think doing this is similar to documenting how `html!` uses `VNode`.

That is something we're planning to do (and already started doing at some point). The problem there is that Yew's `virtual_dom` is nowhere near as stable yet and a lot of users don't fully understand it (which is obviously due to the lack of documentation) so there aren't that many people even capable of writing it.